misc/python/materialize/checks/uuid.py

```diff
@@ -0,0 +1,74 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+from textwrap import dedent
+from typing import List
+
+from materialize.checks.actions import Testdrive
+from materialize.checks.checks import Check
+from materialize.util import MzVersion
+
+
+class UUID(Check):
+    def initialize(self) -> Testdrive:
+        return Testdrive(
+            dedent(
+                """
+            > CREATE TABLE uuid_table (f1 UUID, f2 UUID, f3 STRING);
+            > INSERT INTO uuid_table VALUES (uuid_generate_v5('a0eebc99-9c0b-4ef8-bb6d-6bb9bd380a11', 'bar'), 'a0eebc99-9c0b-4ef8-bb6d-6bb9bd380a12', 'baz');
+            """
+                if self.base_version >= MzVersion(0, 46, 0)
```

Comment:
This conditional is being repeated several times in the file. Since going forward most uses of the version will be of the "skip entire check" variety, can we come up with a more concise syntax:

```
@skip_check_if_version_less_than(some_version_goes_here)
class UUID(Check):
....
```
or 
```
class UUID(Check):
    def requires(self):
         return self.base_version >= MzVersion(0, 46, 0)
```
or something along those lines, whichever syntax you feels more natural to you.